packages/sdk/src/chains/EVM/utils/depositFns.ts

```diff
@@ -0,0 +1,296 @@
+  recipientAddress,
+  tokenInstance,
+  bridgeInstance,
+  provider,
+  handlerAddress,
+  domainId,
+  resourceId,
+  feeData,
+  confirmations,
+  overrides,
+}: Erc721TransferParamsType): Promise<ContractReceipt> => {
+  // construct the deposit data
+  const depositData = createERCDepositData(tokenId, 20, recipientAddress);
+
+  // Chcke approval for this particular tokenID
+  console.log(
+    'Approval before deposit',
+    await getApproved(Number(tokenId), tokenInstance, handlerAddress),
+  );
+
+  // pass data to smartcontract function and create a transaction
+  return executeDeposit(
+    domainId,
+    resourceId,
+    depositData,
+    feeData,
+    bridgeInstance,
+    confirmations,
+    provider,
+    overrides,
+  );
+};
+
+/**
+ * Executes a deposit operation using the specified parameters and returns a contract receipt.
+ *
+ * @example
+ * const domainId = '1';
+ * const resourceId = '0x1234567890123456789012345678901234567890';
+ * const depositData = '0xabcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789';
+ * const feeData = { type: 'basic', fee: ethers.BigNumber.from('1000000000000000'), feeData: {} }; // fee data is madatory
+ * const bridgeInstance = new Bridge() // Bridge instance from sygma-contracts package;
+ * const confirmations = 3;
+ * const provider = new ethers.providers.JsonRpcProvider();
+ * const overrides = { gasLimit: 200000 };
+ *
+ * executeDeposit(domainId, resourceId, depositData, feeData, bridgeInstance, confirmations, provider, overrides)
+ *   .then((receipt) => console.log('Deposit executed:', receipt))
+ *   .catch((error) => console.error('Error on deposit execution:', error));
+ *
+ * @param {string} domainId - The unique identifier for destination network.
+ * @param {string} resourceId - The resource ID associated with the token.
+ * @param {string} depositData - The deposit data required for the operation.
+ * @param {FeeDataResult} feeData - The fee data result for the deposit operation.
+ * @param {Bridge} bridgeInstance - The bridge instance used to perform the deposit operation.
+ * @param {number} confirmations - The number of confirmations required before the transaction is considered successful.
+ * @param {providers.Provider} provider - The provider used for the Ethereum network connection.
+ * @param {ethers.PayableOverrides} [overrides] - Optional transaction overrides to be applied.
+ * @returns {Promise<ContractReceipt>} A promise that resolves to a contract receipt once the deposit is executed.
+ */
+export const executeDeposit = async (
+  domainId: string,
+  resourceId: string,
+  depositData: string,
+  feeData: FeeDataResult,
+  bridgeInstance: Bridge,
+  confirmations: number,
+  provider: providers.Provider,
+  overrides?: ethers.PayableOverrides,
+): Promise<ContractReceipt> => {
+  try {
+    const gasPrice = await isEIP1559MaxFeePerGas(provider);
+    const gasPriceStringify = typeof gasPrice !== 'boolean' ? gasPrice.toString() : undefined;
```

Comment:
Isn't gasPrice always BigNumber and you can keep it as BigNumber?

Reply:
That code was built for celo network, that didn't support EIP1559 at this time